examples/api/lib/widgets/transitions/listenable_builder.0.dart

```diff
@@ -0,0 +1,168 @@
+// Copyright 2014 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+/// Flutter code sample for [ListenableBuilder].
+
+import 'package:flutter/material.dart';
+
+void main() => runApp(const ListenableBuilderExample());
+
+/// This widget listens for changes in the focus state of the subtree defined by
+/// its [child] widget, changing the border and color of the container it is in
+/// when it has focus.
+///
+/// A [FocusListenerContainer] swaps out the [BorderSide] of a border around the
+/// child widget with [focusedSide], and the background color with
+/// [focusedColor], when a widget that is a descendant of this widget has focus.
+class FocusListenerContainer extends StatefulWidget {
+  const FocusListenerContainer({
+    super.key,
+    required this.child,
+    this.border,
+    this.padding,
+    this.focusedSide,
+    this.focusedColor = Colors.black12,
+  });
+
+  /// This is the border that will be used when not focused, and which defines
+  /// all the attributes except for the [OutlinedBorder.side] when focused.
+  final OutlinedBorder? border;
+
+  /// This is the [BorderSide] that will be used for [border] when the [child]
+  /// subtree is focused.
+  final BorderSide? focusedSide;
+
+  /// This is the [Color] that will be used as the fill color for the background
+  /// of the [child] when a descendant widget is focused.
+  final Color? focusedColor;
+
+  /// The padding around the inside of the container.
+  final EdgeInsetsGeometry? padding;
+
+  /// This is defines the subtree to listen to for focus changes.
+  final Widget child;
+
+  @override
+  State<FocusListenerContainer> createState() => _FocusListenerContainerState();
+}
+
+class _FocusListenerContainerState extends State<FocusListenerContainer> {
+  final FocusNode _focusNode = FocusNode();
+
+  @override
+  void dispose() {
+    _focusNode.dispose();
+    super.dispose();
+  }
+
+  @override
+  Widget build(BuildContext context) {
+    final OutlinedBorder effectiveBorder = widget.border ?? const RoundedRectangleBorder();
+    return ListenableBuilder(
+      listenable: _focusNode,
+      child: widget.child,
```

Comment:
Technically, you could also just use `widget.child` directly in the builder in this case. Since the child is already "pre-built", using the child argument does not have any advantages. No disadvantages either.